controller/service.rst

```diff
@@ -23,17 +23,19 @@ These are the main **advantages** of defining controllers as services:
   service container configuration. This is useful when developing reusable bundles;
 * Your controllers are more "sandboxed". By looking at the constructor arguments,
   it's easy to see what types of things this controller may or may not do;
+* If you're not passing some required dependencies or if you are injecting some
+  non-existent services, you'll get errors during the container compilation
+  instead of during runtime execution;
 * Since dependencies must be injected manually, it's more obvious when your
   controller is becoming too big (i.e. if you have many constructor arguments).
 
 These are the main **drawbacks** of defining controllers as services:
 
-* It takes more work to create the controllers because they don't have
-  automatic access to the services or to the base controller shortcuts;
+* It takes more work to create the controllers and they become more verbose
```

Comment:
I'm sorry but I don't understand your comment. Let me show you why I wrote that phrase.

---

For such a common thing as rendering a template, if you use the base controller, you do this:

```php
return $this->render('homepage.html.twig');
```

Just 1 short and beautiful PHP line.

However, if you don't use that base controller, the equivalent code is this one:

```php
        if (!$this->container->has('twig')) {
            throw new \LogicException('...');
        }
        
        $response = new Response();
        $response->setContent($this->container->get('twig')->render('homepage.html.twig'));

        return $response;
```

--- 

So, I think the previous phrase is correct: "It takes more work" ... and "they become more verbose".